### ISSUE TITLE
add repmat node

### ISSUE THREAD

**Issue by jgillis** [casadi/casadi#1527]

**Comment by jaeandersson** [comment]
The adjoint would be some kind of summation of the matrix blocks I guess.

**Comment by jgillis** [comment]
Yes, I called it Repsum

**Comment by jaeandersson** [comment]
i would define the repsum node with in-place addition: Z = X + repsum(Y). Sparsity of Z would be given by X. Cf. how multiplication is defined.

**Comment by jgillis** [comment]
Changing ticket title for:
  * [ ] allow non-repeated inputs
  * [ ] make repsum in-place

**Comment by jgillis** [comment]
I just realised that Kronencker #939 , can be seen as a generalization of repmat.
It is perhaps a better primitive to have than repmat.

**Comment by jaeandersson** [comment]
> It is perhaps a better primitive to have than repmat

Good call. Kronencker indeed interesting generalization. Repmat is simply: `kron(ones(n,m),A)`.
